modules/mockk/src/commonMain/kotlin/io/mockk/MockK.kt

```diff
@@ -1,8 +1,70 @@
+import kotlin.reflect.KClass
+
+
+
+private val nonMockableClasses = listOf(
+
+  System::class,
+
+  java.io.File::class,
+
+  java.nio.file.Path::class,
+
+  // Add more classes as necessary
+
+)
+
+
+
 @file:Suppress("NOTHING_TO_INLINE")
 
+
+
 package io.mockk
 
-import kotlin.reflect.KClass
+
+
+inline fun <reified T : Any> mockk(
```

Comment:
Also, this is just a wrapper for the `MockKDsl` class, I don't think it should contain any business logic.

Can you please move the check to `AbstractMockFactory`?

Reply:
SURE!!!